core/authorship/block_builder.hpp

```diff
@@ -21,18 +21,17 @@ namespace kagome::basic_authorship {
 
     /**
      * Push extrinsic to wait its inclusion to the block
-     * Returns result containing true if xt was pushed, false otherwise and
-     * failure if error happened
+     * Returns result containing success if xt was pushed, error otherwise
      */
-    virtual outcome::result<bool> pushExtrinsic(
+    virtual outcome::result<void> pushExtrinsic(
         const primitives::Extrinsic &extrinsic) = 0;
 
     /**
      * Create a block from extrinsics and header
      */
-    [[nodiscard]] virtual primitives::Block bake() const = 0;
+    [[nodiscard]] virtual outcome::result<primitives::Block> bake() const = 0;
```

Comment:
why nodiscard? outcome::result is already nodiscard

Reply:
clang-tidy suggests so